subsys/net/ip/net_context.c

```diff
@@ -2401,53 +2401,68 @@ static int recv_udp(struct net_context *context,
 
 	ARG_UNUSED(timeout);
 
+	/* If the context already has a connection handler, it means it's
+	 * already registered. In that case, all we have to do is 1) update
+	 * the callback registered in the net_context and 2) update the
+	 * user_data and remote address and port using net_conn_update().
+	 *
+	 * The callback function passed to net_conn_update() must be the same
+	 * function as the one passed to net_conn_register(), not the callback
+	 * set for the net context passed by recv_udp().
+	 */
 	if (context->conn_handler) {
-		net_conn_unregister(context->conn_handler);
-		context->conn_handler = NULL;
-	}
+		context->recv_cb = cb;
+		ret = net_conn_update(context->conn_handler,
+				      net_context_packet_received,
+				      user_data,
+				      context->flags & NET_CONTEXT_REMOTE_ADDR_SET ?
+						&context->remote : NULL,
+				      ntohs(net_sin(&context->remote)->sin_port));
+	} else {
 
```

Comment:
The else branch is certainly one option, it is just that there is lot of code to indent which makes the lines longer. We could avoid the else branch like it was originally. For example like this:
```
if (context->conn_handler != NULL) {
	context->recv_cb = cb;
	ret = net_conn_update(context->conn_handler,
			      net_context_packet_received,
			      user_data,
			      context->flags & NET_CONTEXT_REMOTE_ADDR_SET ?
					&context->remote : NULL,
			      ntohs(net_sin(&context->remote)->sin_port));
	return ret;
}
```

Reply:
Okay, fixed.